devenv/aws-setup/initialize.py

```diff
@@ -0,0 +1,72 @@
+import boto3
```

Comment:
I intentionally didn't add tests to this file:
1. To add testing there would need to be another python file which would bloat the `devenv` code
2. The biggest pitfall of this will be if `cdk` creates a template that changed formats, at which point we'll need to significantly change this file anyway. We can't be resilient to that, but also it shouldn't change unless there's a full new `aws-cdk` version.

Reply:
Fair enough. However, the file would still be much more readable if it was structured in a more testable way, even though we don't have to necessarily test it.

I'd love to see functions defined for most of the tasks, and having all of the functions being called from a top-level main() function residing in a `if __name__ == '__main__'` block.

That would make the code more easy to follow, and experiment with if I'd like to import the module from a shell and call a single function for example.

Example:
```python
def main():
    client = boto3.client(...)
    template, template_resource_ids_for_type = read_cdk_template()
    setup_tables(...)
    setup_functions(...)
    output_cdk_template(...)
    
if __name__ == "__main__":
    main()
```

What would you think of a structure like that?

Reply:
I disagree on this one, but not enough to fight it if someone else agrees with you.

Python is ultimately a scripting language that ended up getting a lot of users because of how easy it is to use and the libraries that support it.

This here is just a bash script written in python; you wouldn't expect there to be a `main` in a bash script, and I would argue you wouldn't want there to be a `main` here because it's a one off script and should be structured like one. 

If this were the lambda written in python or a test suite I'd be 1000% with you, but I think the for the purpose this script has it's actually better to have this clearly be a script and not a program.

Reply:
My 2 cents: being able to reason about separate operations is the important part. You have comments that help with that, so it's reasonably clear already, but it would be better to separate those as functions. I'd say the same thing about structuring bash scripts (although for bash it's largely aspirational, I sometimes fail to follow it myself). And I don't care about the `if __name__ == '__main__':` block one way or the other.

Reply:
> Python is ultimately a scripting language that ended up getting a lot of users because of how easy it is to use and the libraries that support it.

I'd argue that a lot of the success of Python comes from their philosophy which strongly emphasizes code readability. Yes this is "just a bash script", but when Python gives us the ability to structure the code in a readable and more maintainable way at basically no cost - why not do it?

Given that code is read more times than it's written, optimizing our code for readability will save us time in the long run.

Reply:
> If this were the lambda written in python or a test suite I'd be 1000% with you, but I think the for the purpose this script has it's actually better to have this clearly be a script and not a program.

I'm curious about your reasoning here. Even for a script, what are the benefits of writing everything as a flat sequence of instructions as opposed to organize it into functions?